src/ILVerify/src/ProcessedMethodTracker.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection.Metadata;
+using System.Linq;
+using Internal.TypeSystem;
+using Internal.TypeSystem.Ecma;
+using static System.Console;
+
+namespace ILVerify
+{
+    internal class ProcessedMethodTracker
+    {
+        private class MethodResult
+        {
+            public MethodDesc Method;
+            public string Name;
+            public bool Verified;
+        };
+        private class ModuleResult {
+            public IList<MethodResult> Methods = new List<MethodResult>();
```

Comment:
The list and dictionary are not needed now. It can be a simple counter.

Reply:
I replaced the list with simple counters, but kept the dictionary, because it holds them per module.

Reply:
All methods in one module are always verified together, so the tracker will be always counting the method for just one module. Is the separate tracker really worth it - can the VerifyAssembly method look like this?

```
private void VerifyAssembly(PEReader peReader, EcmaModule module, string path)
{
    int numErrors = 0;
    int verifiedMethodCounter = 0;
    int methodCounter = 0;

    MetadataReader metadataReader = peReader.GetMetadataReader();
    foreach (var methodHandle in metadataReader.MethodDefinitions)
    {
        // get fully qualified method name
        var methodName = GetQualifiedMethodName(metadataReader, methodHandle);

        bool verifying = ShouldVerifyMethod(methodName);
        if (_verbose)
        {
            Write(verifying ? "Verifying " : "Skipping ");
            WriteLine(methodName);
        }

        if (verifying)
        {
            var results = _verifier.Verify(peReader, methodHandle);
            foreach (var result in results)
            {
                PrintResult(result, module, path);
                numErrors++;
            }

            verifiedMethodCounter++;
        }

        methodCounter++;
    }

    if (_printStatistics)
    {
        WriteLine($"Methods found: {methodCounter}");
        WriteLine($"Methods verified: {verifiedMethodCounter}");
    }

    if (numErrors > 0)
        WriteLine(numErrors + " Error(s) Verifying " + path);
    else
        WriteLine("All Classes and Methods in " + path + " Verified.");
}
```